test/integration/change-streams/change_stream.test.ts

```diff
@@ -1563,3 +1493,584 @@ describe('Change Streams', function () {
     });
   });
 });
+
+describe('ChangeStream resumability', function () {
+  let client: MongoClient;
+  let collection: Collection;
+  let changeStream: ChangeStream;
+  let aggregateEvents: CommandStartedEvent[] = [];
+
+  const changeStreamResumeOptions: ChangeStreamOptions = {
+    fullDocument: 'updateLookup',
+    collation: { locale: 'en', maxVariable: 'punct' },
+    maxAwaitTimeMS: 20000,
+    batchSize: 200
+  };
+
+  const resumableErrorCodes = [
+    { error: 'HostUnreachable', code: 6 },
+    { error: 'HostNotFound', code: 7 },
+    { error: 'NetworkTimeout', code: 89 },
+    { error: 'ShutdownInProgress', code: 91 },
+    { error: 'PrimarySteppedDown', code: 189 },
+    { error: 'ExceededTimeLimit', code: 262 },
+    { error: 'SocketException', code: 9001 },
+    { error: 'NotWritablePrimary', code: 10107 },
+    { error: 'InterruptedAtShutdown', code: 11600 },
+    { error: 'InterruptedDueToReplStateChange', code: 11602 },
+    { error: 'NotPrimaryNoSecondaryOk', code: 13435 },
+    { error: 'StaleShardVersion', code: 63 },
+    { error: 'StaleEpoch', code: 150 },
+    { error: 'RetryChangeStream', code: 234 },
+    { error: 'FailedToSatisfyReadPreference', code: 133 },
+    { error: 'CursorNotFound', code: 43 }
+  ];
+
+  const is4_2Server = (serverVersion: string) =>
+    gte(serverVersion, '4.2.0') && lt(serverVersion, '4.3.0');
+
+  beforeEach(async function () {
+    client = this.configuration.newClient({ monitorCommands: true });
+    client.on('commandStarted', filterForCommands(['aggregate'], aggregateEvents));
+    await client.connect();
+    // 3.6 servers do not support creating a change stream on a database that doesn't exist
+    await client
+      .db('resumabilty_tests')
+      .dropDatabase()
+      .catch(e => e);
+    collection = await client.db('resumabilty_tests').createCollection('foo');
+  });
+
+  afterEach(async function () {
+    await changeStream.close();
+    await client.close();
+    aggregateEvents = [];
+  });
+
+  context('iterator api', function () {
+    context('#next', function () {
+      for (const { error, code } of resumableErrorCodes) {
+        it(
+          `resumes on error code ${code} (${error})`,
+          { requires: { topology: '!single', mongodb: '>=4.2' } },
+          async function () {
+            changeStream = collection.watch([]);
+            await initIteratorMode(changeStream);
+
+            await client.db('admin').command(<FailPoint>{
+              configureFailPoint: is4_2Server(this.configuration.version)
+                ? 'failCommand'
+                : 'failGetMoreAfterCursorCheckout',
+              mode: { times: 1 },
+              data: {
+                failCommands: ['getMore'],
+                errorCode: code
+              }
+            });
+
+            await collection.insertOne({ name: 'bailey' });
```

Comment:
is this a sneaky way to immortalize yourself in the codebase? 😆

Reply:
caught red handed! 😱 

I usually do this for testing and must not have noticed when I pushed.  Should I use a number instead of someones name?

Reply:
I mean I don't really care, I was just amused, up to you if you want to tie your reputation to this test 😄

Reply:
I'll leave it.  fwiw I usually rotate between all our names, so perhaps in the future you'll see a commit from me that has `{name: 'daria' }` or `{name: 'neal'}` or `{ name: 'durran' }`